src/state-client.ts

```diff
@@ -284,101 +323,144 @@ class StateClient {
-    this.signedRequest(identifier, 'GET', ServerState.USER, undefined, true);
+    this.signedRequest<UserInfo>(identifier, 'GET', ServerState.USER, undefined, true);
 
   /** @ignore */
-  public getViewerState = (identifier: string) => this.getState(identifier, ServerState.VIEWER);
+  public getViewerState = (identifier: string) => this.getState<Object>(identifier, ServerState.VIEWER);
 
   /** @ignore */
-  public getExtensionViewerState = (identifier: string) => this.getState(identifier, ServerState.EXTENSION_VIEWER);
+  public getExtensionViewerState = (identifier: string) => this.getState<Object>(identifier, ServerState.EXTENSION_VIEWER);
 
   /** @ignore */
-  public getExtensionSecretState = (identifier: string) => this.getState(identifier, ServerState.EXTENSION_SECRET);
+  public getExtensionSecretState = (identifier: string) => this.getState<Object>(identifier, ServerState.EXTENSION_SECRET);
 
   /** @ignore */
-  public getChannelState = (identifier: string) => this.getState(identifier, ServerState.CHANNEL);
+  public getChannelState = <StateType = unknown>(identifier: string) => this.getState<StateType>(identifier, ServerState.CHANNEL);
 
   /** @ignore */
-  public getExtensionState = (identifier: string) => this.getState(identifier, ServerState.EXTENSION);
+  public getServerConfig = <ResponseType = unknown>(identifier: string) => this.getConfig<ResponseType>(identifier, ServerConfig.ALL);
 
   /** @ignore */
-  public setViewerState = (identifier: string, state: any) =>
-    this.postState(identifier, ServerState.VIEWER, JSON.stringify(state));
+  public getChannelConfig = <ResponseType = unknown>(identifier: string) => this.getConfig<ResponseType>(identifier, ServerConfig.CHANNEL);
 
   /** @ignore */
-  public setExtensionViewerState = (identifier: string, state: any) =>
-    this.postState(identifier, ServerState.EXTENSION_VIEWER, JSON.stringify(state));
+  public getExtensionConfig = <ResponseType = unknown>(identifier: string) => this.getConfig<ResponseType>(identifier, ServerConfig.EXTENSION);
+
+  /** @ignore */
+  public getExtensionState = (identifier: string) => this.getState<Object>(identifier, ServerState.EXTENSION);
+
+  /** @ignore */
+  public setViewerState = <DataType = unknown, ResponseType = unknown>(identifier: string, state: DataType) =>
+    this.postState<DataType, ResponseType>(identifier, ServerState.VIEWER, state);
+
+  /** @ignore */
+  public setExtensionViewerState = <DataType = unknown, ResponseType = unknown>(identifier: string, state: DataType) =>
+    this.postState<DataType, ResponseType>(identifier, ServerState.EXTENSION_VIEWER, state);
 
   /** @ignore */
   public patchExtensionViewerState = (identifier: string, multiState: any) =>
-    this.signedRequest(identifier, 'PATCH', 'extension_viewer_state', JSON.stringify(multiState));
+    this.signedRequest<any, Object>(identifier, 'PATCH', 'extension_viewer_state', JSON.stringify(multiState));
```

Comment:
`any`, `Object`, and `JSON.stringify` here. This function can be more strongly typed if possible.

Reply:
So you want the entire project updated, and not just the config stuff now?

Reply:
Update anything that breaks with the new code changes or if you already updating it (as here) go ahead and do it correctly.

The `JSON.stringify` probably breaks stuff as I mentioned in another comment, and if you have an `unknown` type, use that instead of `any`. In addition, `Object` is not the proper return type. And as we discussed, if you are leaving something untyped for a later fix, you need to add a `// TODO: Types need to be corrected` style comment.